x-pack/legacy/plugins/apm/public/components/app/APMIndicesPermission/index.tsx

```diff
@@ -40,19 +40,21 @@ export const APMIndicesPermission: React.FC = ({ children }) => {
     return null;
   }
 
-  const indicesWithoutPermission = Object.keys(indicesPrivileges).filter(
-    index => !indicesPrivileges[index].read
-  );
-
-  // Show permission warning when a user has at least one index without Read privilege,
-  // and he has not manually dismissed the warning
-  if (!isEmpty(indicesWithoutPermission) && !isPermissionWarningDismissed) {
-    return (
-      <PermissionWarning
-        indicesWithoutPermission={indicesWithoutPermission}
-        onEscapeHatchClick={() => setIsPermissionWarningDismissed(true)}
-      />
+  if (!isEmpty(indicesPrivileges)) {
```

Comment:
nit: what do you think about having the API return something more explicit about whether the appropriate permissions are available? Feels like it's more clear then where the responsibility lies about who decides if that is the case. Right now there's some logic in the server and some in the client. Something like:

```
{
  sufficient_privileges: false,
  missing_privileges: {
	'apm_oss.transactionIndices': 'apm-7.5.1-transaction-*'
  }
}
```

Reply:
yeah, that sounds good.

Reply:
Instead of introducing `sufficient_privileges` can't we use `has_all_requested` which is already available on the client?

Reply:
Btw. doesn't the current logic already handle the case where `indicesPrivileges` is empty?

`Object.keys(indicesPrivileges)` becomes `[]` and the execution will proceed to `return <>{children}</>;`

Reply:
> Instead of introducing `sufficient_privileges` can't we use `has_all_requested` which is already available on the client?

@dgieselaar / @sqren, In that case, I won't return the `missing_privileges`, I will return the entire response that I got from calling `/_security/user/_has_privileges` which is:
```js
interface IndexPrivileges {
  has_all_requested: boolean;
  username: string;
  index: Record<string, { read: boolean }>;
}
```

Then the client will be responsible, as it already is, to loop through the indices to show it on the page.

Reply:
Sounds good to me 👍 
@dgieselaar did you have something else in mind?

Reply:
That should be fine, my idea was to create a map on the server for config key/value so the user can debug their config (either `kibana.yml` or the UI indices), not sure if we can do that on the client. But it's just a small improvement.

Reply:
Based on the current response, the client will be able to get an index privilege by running:
```
const index = indices['apm_oss. transactionIndices'];
indicesPrivileges.index[index]
```

Reply:
@cauemarcondes Is `indices` available on the client then? And will that work in NP?

Reply:
No, it's not. But when it is we can do as I said. I don't think it's responsibility of this PR to make this change too.